cohttp-lwt/src/client.ml

```diff
@@ -3,7 +3,7 @@ module Header = Cohttp.Header
 
 exception Retry
 
-module Make (IO : S.IO) (Net : S.Net with module IO = IO) (Time : Mirage_time.S) = struct
+module Make (IO : S.IO) (Net : S.Net with module IO = IO) (Sleep : S.Sleep) = struct
```

Comment:
@dinosaure this should be fine for functoria, no?

Reply:
Hmmhmm, I'm not sure. I mean, for `mirage`/`functoria`, we already have a mapping between a `time` device and the `module type Mirage_time.S`. You can do that but that mostly means that CoHTTP devs must keep in their mind that this interface should **always** be equal to `Mirage_time.S` (and on the other side, mirage devs should be aware that any upgrade of `Mirage_time.S` implies an update of `Cohttp_lwt.S.Sleep`).

If we keep some packages as interfaces (like `mirage-time`, `mirage-block`, etc.), it's to be sure to provide a coherent ecosystem and a simple update of these interfaces should be systematic along our `mirage/*` ecosystem.

Then, the release cycle of `Mirage_time.S` is not a real problem - I mean, the interface did not change for a long time. About dependencies, I still think that it's fine. `mirage-time` is just an interface which can be used across projects and, again, ensure a _certain_ homogeneity across our multi-repositories organization. I'm not sure that anyone want to take this hidden responsibility between what the MirageOS expects and what CoHTTP wants.

Reply:
note that the application of the `Cohttp_lwt.Make` functor in `cohttp-mirage/src/client.ml:76` enforces compatibility between `Cohttp_lwt.S.Sleep` and `Mirage_time.S`.

Nevertheless _I_ would prefer using `Mirage_time.S` in `Cohttp_lwt.Client`, adding dependency on `mirage-time`.